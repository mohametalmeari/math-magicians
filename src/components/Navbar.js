import { NavLink } from 'react-router-dom';

const links = [
  { path: '/', text: 'Home' },
  { path: 'calculator', text: 'Calculator' },
  { path: 'quote', text: 'Quote' },
];
const Navbar = () => (
  <nav>
    <h1>
      Math Magicians
    </h1>
    <ul className="menu">
      {links.map((link) => (
        <li className="page-link" key={link.text}>
          <NavLink
            to={link.path}
          >
            {link.text}
          </NavLink>
        </li>
      ))}
    </ul>
  </nav>
);
export default Navbar;
