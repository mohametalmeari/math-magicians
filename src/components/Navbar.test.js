import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Navbar from './Navbar';

describe('Navbar Component', () => {
  test('test render of navbar', () => {
    // extract the container
    const { container } = render(
      // Wrapping <Navbar /> inside <MemoryRouter /> to simulate <BrowserRouter />
      // due to the existence of <NavLink /> inside Navbar component.
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });

  test('test render of navigation links', () => {
    const links = [
      { path: '/', text: 'Home' },
      { path: '/calculator', text: 'Calculator' },
      { path: '/quote', text: 'Quote' },
    ];

    // extract the getByText method that returns an element
    const { getByText } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    );

    links.forEach((link) => {
      const navLink = getByText(link.text);
      expect(navLink.getAttribute('href')).toBe(link.path);
    });
  });
});
