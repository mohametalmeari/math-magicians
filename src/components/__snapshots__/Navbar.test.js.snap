// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component test render of navbar 1`] = `
<div>
  <nav>
    <h1>
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        Math Magicians
      </a>
    </h1>
    <ul
      class="menu"
    >
      <li
        class="page-link"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="page-link"
      >
        <a
          class=""
          href="/calculator"
        >
          Calculator
        </a>
      </li>
      <li
        class="page-link"
      >
        <a
          class=""
          href="/quote"
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
